dap_prinz_green_jobs/pipeline/ojo_application/ojo_sample/deduplication_utils.py

```diff
@@ -0,0 +1,139 @@
+"""
+Functions to deal with duplication in job adverts.
+
+get_deduplicated_job_adverts takes the job adverts (only job id and date columns are needed)
+and chunks up the dates and then removes duplicates from each chunk, before recombining.
+
+"""
+
+from datetime import datetime, timedelta, date
+from dateutil.relativedelta import relativedelta
+import random
+from hashlib import md5
+
+from dap_prinz_green_jobs import logger
+
+
+def short_hash(text):
+    """Generate a unique short hash for this string - from ojd_daps"""
+    hx_code = md5(text.encode()).hexdigest()
+    int_code = int(hx_code, 16)
+    short_code = str(int_code)[:16]
+    return int(short_code)
+
+
+def get_date_ranges(start, end, num_units=7, unit_type="days"):
```

Comment:
ditto types for the rest of the functions - but your doc strings have that context so up to you!